src/ifcgeom/infra_sweep_helper.cpp

```diff
@@ -143,28 +167,203 @@ taxonomy::loft::ptr ifcopenshell::geometry::make_loft(const Settings& settings_,
                     } else if (rotation_a != rotation_b) {
 						Logger::Error("Direction vectors on cross section placements only supported when used consistently");
 					}
+
 					taxonomy::loop::ptr w1, w2;
 					taxonomy::edge::ptr e1, e2;
+                    taxonomy::point3::ptr p1, p2;
+
 					for (auto tmp_ : boost::combine(loops_a, loops_b)) {
 						boost::tie(w1, w2) = tmp_;
-						if (w1->children.size() != w2->children.size()) {
-							Logger::Warning("Mismatching number of edges: " +
-								std::to_string(w1->children.size()) + " vs " +
-								std::to_string(w2->children.size()),
-								inst
-							);
+
+						if (w1->closed != w2->closed) {
+							Logger::Warning("Mismatching closed property on loops", inst);
+							return nullptr;
+                        }
+
+						if (w1->tags.is_initialized() != w2->tags.is_initialized()) {
+							Logger::Warning("Mismatching availability tags on loops", inst);
 							return nullptr;
+                        }
+
+						if (w1->tags) {
+							// check uniqueness
+                            std::set<std::string> tags_seen;
+                            for (const auto& t : *w1->tags) {
+								if (tags_seen.find(t) != tags_seen.end()) {
+									Logger::Warning("Duplicate tag '" + t + "' on loft profile", inst);
```

Comment:
The example in https://github.com/bSI-InfraRoom/IFC-infra-unit-test/pull/106 suggests that there can be duplicate Tag values. This may work for branching, but it doesn't for string lines. Unless specifically defined somewhere, Tag values should be unique.

Reply:
I had multiset initially. So I'm open to exploring that further. But I'm unsure if it changes much given how currently things are handled because the current code only cares for set intersections which would be the same for set vs multiset.